Packages/Netherlands3D/glTFast/Runtime/Scripts/Material/ShaderGraphMaterialGenerator.cs

```diff
@@ -0,0 +1,634 @@
+        // /// <summary>Shader property ID for property transmissionTexture_texCoord</summary>
+        // public static readonly int TransmissionTextureUVChannelProperty = Shader.PropertyToID("transmissionTexture_texCoord");
+
+#if USING_HDRP_10_OR_NEWER || USING_URP_12_OR_NEWER
+        // ReSharper disable MemberCanBeProtected.Global
+        // const string KW_DISABLE_DECALS = "_DISABLE_DECALS";
+        /// <summary>Shader keyword _DISABLE_SSR_TRANSPARENT</summary>
+        public const string DisableSsrTransparentKeyword = "_DISABLE_SSR_TRANSPARENT";
+        /// <summary>Shader keyword _ENABLE_FOG_ON_TRANSPARENT</summary>
+        public const string EnableFogOnTransparentKeyword = "_ENABLE_FOG_ON_TRANSPARENT";
+        /// <summary>Shader keyword _SURFACE_TYPE_TRANSPARENT</summary>
+        public const string SurfaceTypeTransparentKeyword = "_SURFACE_TYPE_TRANSPARENT";
+
+        /// <summary>Shader pass TransparentDepthPrepass</summary>
+        public const string ShaderPassTransparentDepthPrepass = "TransparentDepthPrepass";
+        /// <summary>Shader pass TransparentDepthPostpass</summary>
+        public const string ShaderPassTransparentDepthPostpass = "TransparentDepthPostpass";
+        /// <summary>Shader pass TransparentBackface</summary>
+        public const string ShaderPassTransparentBackface = "TransparentBackface";
+        /// <summary>Shader pass RayTracingPrepass</summary>
+        public const string ShaderPassRayTracingPrepass = "RayTracingPrepass";
+        /// <summary>Shader pass DepthOnly</summary>
+        public const string ShaderPassDepthOnlyPass = "DepthOnly";
+
+        /// <summary>Shader property ID for property _AlphaDstBlend</summary>
+        public static readonly int AlphaDstBlendProperty = Shader.PropertyToID("_AlphaDstBlend");
+        /// <summary>Shader property ID for property _ZTestGBuffer</summary>
+        public static readonly int ZTestGBufferProperty = Shader.PropertyToID("_ZTestGBuffer");
+        // ReSharper restore MemberCanBeProtected.Global
+#endif
+
+#if !UNITY_SHADER_GRAPH_12_OR_NEWER
+        static Dictionary<MetallicShaderFeatures,Shader> s_MetallicShaders = new Dictionary<MetallicShaderFeatures,Shader>();
+        static Dictionary<SpecularShaderFeatures,Shader> s_SpecularShaders = new Dictionary<SpecularShaderFeatures,Shader>();
+        static Dictionary<UnlitShaderFeatures,Shader> s_UnlitShaders = new Dictionary<UnlitShaderFeatures,Shader>();
+#else
+        static Shader s_MetallicShader;
+        static Shader s_SpecularShader;
+        static Shader s_UnlitShader;
+#endif
+
+        /// <inheritdoc />
+        protected override Material GenerateDefaultMaterial(bool pointsSupport = false) {
+            if(pointsSupport) {
+                Logger?.Warning(LogCode.TopologyPointsMaterialUnsupported);
+            }
+            var defaultMaterial = GetMetallicMaterial(MetallicShaderFeatures.Default);
```

Comment:
Het resultaat van deze functie (GetMetallicMaterial) kan null zijn, en dan krijg je een console error in de volgende regel. Dit nam ik waar toen ik de sample inlaadde in een nieuw project.

```
NullReferenceException: Object reference not set to an instance of an object
GLTFast.Materials.ShaderGraphMaterialGenerator.GenerateDefaultMaterial (System.Boolean pointsSupport) (at Library/PackageCache/nl.netherlands3d@dd98df3605/glTFast/Runtime/Scripts/Material/ShaderGraphMaterialGenerator.cs:179)
GLTFast.Materials.MaterialGenerator.GetDefaultMaterial (System.Boolean pointsSupport) (at Library/PackageCache/nl.netherlands3d@dd98df3605/glTFast/Runtime/Scripts/Material/MaterialGenerator.cs:239)
```